src/serialization/string_utils.cpp

```diff
@@ -822,9 +822,9 @@ std::vector<std::string> quoted_split(const std::string& val, char c, int flags,
 	return res;
 }
 
-std::pair<int, int> parse_range(const std::string& str)
+std::pair<int, int> parse_range(const std::string& str, bool alter_sep)
 {
-	const std::string::const_iterator dash = std::find(str.begin(), str.end(), '-');
+	const std::string::const_iterator dash = alter_sep ? std::find(str.begin(), str.end(), '_') : std::find(str.begin(), str.end(), '-');
```

Comment:
I think rather than having a switch between the two syntaxes, it would make more sense to try both. Try the new one first, and if that fails, try the old one. If both fail, it's probably just a single number.

By the way, if this is the reason you said it has to be a single character, then you're just using the wrong function. You should use `str.find()` instead if you want to search for a substring. That said, making it be multiple characters would also complicate the logic later on, so I'm not specifically recommending you do this. I'll leave it up to you which is better.

Reply:
except what '-' is also for negative value, i don't like idea to check that for both value and separator for intervalof value.

Reply:
Another possibility would be "don't accept `-` as a separator if it appears elsewhere in the string", so the old syntax with `-` would only be allowed if both numbers were zero-or-more.

Reply:
you'll already find solution in https://github.com/wesnoth/wesnoth/pull/7829, and i don't see how locate good '-' because each of two number can be have one or more characters in string